tests/test_linearize.py

```diff
@@ -611,4 +611,4 @@ def test_different_dtype():
 
     # Check generated code has different strides for different dtypes
     assert "bL0(x,y) b[(x)*y_stride0 + (y)]" in str(op1)
-    assert "L0(x,y) f[(x)*y_stride1 + (y)]" in str(op1)
+    assert "L0(x,y) f[(x)*y_stride0 + (y)]" in str(op1)
```

Comment:
Can we parametrize this test with autopadding so `y_stride1` case is there for "mixed" padding

Reply:
it's not possible by construction anymore; autopadding is now a global choice, in essence. This means that if you have an fp32 and an fp64 the data will be padded to either fp32 or fp64; in the former case, the performance will be suboptimal. But regardless, we will always only generate one stride per dimension. See `types/basic.py::AbstractFunction::__padding_dtype__` logic

That said, I've now parametrized the test as you suggested, but the expected outcome is always `y_stride0`